### PR TITLE
[IMP] delivery_carrier_label_postlogistics: improve test coverage

### PR DESCRIPTION
**Pull request by sebalix**
To be frank I do not understand very well all the features brought by the module, I added tests for some of the computed fields without be sure how they should work.
Green CI: https://travis-ci.org/sebalix/delivery-carrier/builds/456915878

### CODE REVIEW

**Comment by sebalix** [comment]
ping @simahawk

File: delivery_carrier_label_postlogistics/tests/test_delivery.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+class TestDelivery(common.SavepointCase):
+
+    def setUp(self):
+        super(TestDelivery, self).setUp()
+        self.service_group = self.env['postlogistics.service.group'].create({
+            'name': "TEST SERVICE GROUP",
+            'group_extid': 45879089,    # Random
+        })
+        self.carrier = self.env['delivery.carrier'].create({
```
Comment (simahawk) [inline delivery_carrier_label_postlogistics/tests/test_delivery.py]: do you need these records to be re-created for each test? If not use setUpClass pls (and the trick for tracking disable ;))

File: delivery_carrier_label_postlogistics/tests/test_delivery.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            - no available option configured on the carrier
+        """
+        services = self.env['delivery.carrier.template.option'].search(
+            [('postlogistics_type', 'in', SINGLE_OPTION_TYPES)])
+        self.assertEqual(
+            self.carrier.allowed_tmpl_options_ids,
+            services)
+
+    def test_carrier_allowed_tmpl_options_ids_2(self):
+        """Check allowed template options given that there is:
```
Comment (simahawk) [inline delivery_carrier_label_postlogistics/tests/test_delivery.py]: pls, respect https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings
moreover they are printed on tests exec so you should really display only one small line ;)

File: delivery_carrier_label_postlogistics/tests/test_delivery.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+            - no basic template option
+            - one mandatory available option configured on the carrier
+        """
+        self.carrier.available_option_ids += self.create_carrier_option(
+            values={'postlogistics_type': 'resolution',
+                    'mandatory': True,
+                    'carrier_id': self.carrier.id})
+        single_option_types = SINGLE_OPTION_TYPES[:]
```
Comment (simahawk) [inline delivery_carrier_label_postlogistics/tests/test_delivery.py]: [x for x in SINGLE_OPTION_TYPES if x != 'resolution']

**Review by simahawk — CHANGES_REQUESTED** [review]

File: delivery_carrier_label_postlogistics/tests/test_delivery.py (hunk 1 above)
Comment (sebalix) [reply delivery_carrier_label_postlogistics/tests/test_delivery.py]: Yes I need them to be recreated as they are changed in some tests.

Comment (simahawk) [reply delivery_carrier_label_postlogistics/tests/test_delivery.py]: this is not a single transaction case so you should not get the same changes across tests

Comment (sebalix) [reply delivery_carrier_label_postlogistics/tests/test_delivery.py]: Okay! I'm not aware of the different test cases, 'll check that thanks.
By now, it's fixed.

**Comment by sebalix** [comment]
Fixed

**Review by simahawk — APPROVED** [review]

**Review by grindtildeath — COMMENTED** [review]
Test looks good, but not complete. Would be great if you can add a logical test about :
1. create a sale order with the delivery method created in this test and a stockable product
2. confirm
3. Check that mandatory options are applied and not mandatory are available on the picking, and other options are not available.

Would you have time to do this ? Otherwise I'll merge and add a TODO :wink:

**Comment by sebalix** [comment]
@grindtildeath no time ATM, maybe at the end of week but I really can't promise this, so go ahead if you want :)

**Comment by simahawk** [comment]
@grindtildeath  go ahead... TODO comment is fine, we can get back later

**Pull request merged**